backend/app/services/mcp_providers/providers/base.py

```
@@ -0,0 +1,119 @@
+            token: Authentication token for the provider (user's API key)
+            user_name: Current user's username (for providers that need owner identity)
+
+        Returns:
+            Tuple of (servers, error_message). If successful, error_message is None.
+        """
+        from app.services.mcp_providers.core.http_client import MCPProviderHTTPClient
+        from app.services.mcp_providers.core.mapper import DataMapper
+
+        config = self.get_config()
+        client = MCPProviderHTTPClient(config)
+        mapper = DataMapper()
```

Ruff (0.15.2)

[error] Local variable `mapper` is assigned to but never used

Remove assignment to unused variable `mapper`

(F841)